### PR TITLE
test: move crashReporter specs to the main process

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This moves the specs for the crashReporter module to the main process, and substantially refactors them to be more readable & straightforward. It's not quite testing the same thing now—we're starting the crash reporter from the main process instead of from the renderer—but I think that's better actually, since starting the crash reporter from the renderer is kind of weird.

There's no way to shut down the crash reporter, though, so this might affect tests that run after this module? I think that was also the case with the old test suite though.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by nornagon** [comment]
Ah, I see the old version had special code to guarantee that these tests run last.

File: docs/api/crash-reporter.md [hunk 1]
```diff
@@ -119,6 +119,14 @@ Remove a extra parameter from the current set of parameters so that it will not
 
 See all of the current parameters being passed to the crash reporter.
 
+### `crashReporter.getCrashesDirectory()`
+
+Returns `String` - The directory where crashes are temporarily stored before being uploaded.
+
+### `crashReporter.getProductName()`
```
Comment (miniak) [inline docs/api/crash-reporter.md]: I am not sure we want to make this a public API

Comment (miniak) [reply docs/api/crash-reporter.md]: I think it was only mean to be used by the tests

File: spec-main/api-crash-reporter-spec.ts [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+      nodeIntegration: true
+    }
+  })
+  generateSpecs('with sandbox', {
+    webPreferences: {
+      sandbox: true,
+      preload: path.join(fixtures, 'module', 'preload-sandbox.js')
+    }
+  })
+  generateSpecs('with remote module disabled', {
```
Comment (miniak) [inline spec-main/api-crash-reporter-spec.ts]: I don't think we need this anymore

File: docs/api/crash-reporter.md (hunk 1 above)
Comment (happys33) [reply docs/api/crash-reporter.md]: miniak  u are great am new here thx for ur help >  that guys name nornagon and Shelley Vohr Jeremy Apthorp and one  googleapis/nodejs remove my job coz its there pic and madia voice ... i dont know how stop that can u help me .. u can check main and see

Comment (nornagon) [reply docs/api/crash-reporter.md]: In that case can we delete the function? It doesn't seem to be used by anything...

Comment (miniak) [reply docs/api/crash-reporter.md]: I would delete them

**Pull request merged**